crates/composer/src/sound.rs

```diff
@@ -0,0 +1,70 @@
+use cpal::{
+    traits::{DeviceTrait, HostTrait, StreamTrait},
+    BufferSize, SampleRate, StreamConfig,
+};
+use eyre::Result;
+use fundsp::hacker::*;
+
+pub struct SoundController {
+    _stream: cpal::Stream,
+    custom_organ_hz: Shared<f64>,
+    _frontend_net: Net64,
+}
+
+impl SoundController {
+    pub fn new() -> Result<Self> {
+        let host = cpal::default_host();
+
+        let device = host.default_output_device().expect("Failed to find a default output device");
+
+        // TODO(bschwind) - Hardcode this for now, but let's extract these param
+        //                  from device.default_output_config later.
+        let stream_config = StreamConfig {
+            channels: 1,
+            sample_rate: SampleRate(48_000),
+            buffer_size: BufferSize::Default,
+        };
+
+        let err_fn = |err| eprintln!("an error occurred on stream: {}", err);
+
+        let custom_organ_hz = shared(0.0f64);
+        let (frontend_net, mut backend) = Self::build_dsp_graph(&custom_organ_hz);
+
+        let stream = device.build_output_stream(
+            &stream_config,
+            move |data: &mut [f32], _: &cpal::OutputCallbackInfo| {
+                for sample in data {
+                    *sample = backend.get_mono() as f32;
+                }
+            },
+            err_fn,
+            None,
+        )?;
+
+        stream.play()?;
+
+        Ok(Self { _stream: stream, custom_organ_hz, _frontend_net: frontend_net })
+    }
+
+    fn build_dsp_graph(custom_organ_hz: &Shared<f64>) -> (Net64, Net64Backend) {
```

Comment:
Maybe it's just lack of experience with sound synthesis, but I'm a bit lost as to what this function does (especially since it uses that strange DSL reusing the shift operators and whatnot) so a couple comments would go a long way here.